utils/s3/client.cc

```diff
@@ -307,6 +305,12 @@ future<> client::do_retryable_request(group_client& gc, http::request req, http:
     aws::aws_exception request_ex{aws::aws_error{aws::aws_error_type::OK, aws::retryable::yes}};
     while (true) {
         try {
+            // We need to be able to simulate a retry in s3 tests
+            if (utils::get_local_injector().enter("memtable_flush_should_retry")) {
+                throw aws::aws_exception(aws::aws_error{aws::aws_error_type::HTTP_UNAUTHORIZED,
+                    "EACCESS fault injected to simulate retry", aws::retryable::no});
```

Comment:
"Flush" is not a property of s3 client and the connection between this place and sstables flush in replica is not clear. It's better to name the injection "s3_client_fail_authorization" thus making the connection between "failed authorization" and "retried flush" in the test. And the exception text should not mention that it's "flush" either

Also, imagine that someone will want to artificially fail s3 client authorization for some other test. It will need to request for "memtable_flush_should_retry" injection which is (well -- will be) very strange, isn't it.

Reply:
fair, updated.